### DISCUSSION TITLE
there are two processes when typing ps aux|grep code-server

### DISCUSSION THREAD

**Discussion by fly2tomato** [coder/code-server#5241] [Q&A] [answered]
`
ps aux|grep code-server
`

`user        1075  0.2  0.3 714312 57116 pts/1    Sl+  13:48   0:00 /opt/code-server/lib/node /opt/code-server /home/user --disable-telemetry --auth none --bind-addr 0.0.0.0:8080 --cert fa
`
`
user        1087  1.2  0.4 938116 78820 pts/1    Rl+  13:48   0:05 /opt/code-server/lib/node /opt/code-server /home/user --disable-telemetry --auth none --bind-addr 0.0.0.0:8080 --cert fa
`

I am just curious is this normal?

**Comment by jsjoeio** [comment]
Hmm... @code-asher what do you think?

**Comment by code-asher** [comment, marked as answer]
Yup, this is normal.  One of them is the parent of the other.  code-server has a wrapper process so it can restart itself on `SIGUSR2`.

```
$ ps -e -o cmd --forest | grep code-server
/var/tmp/coder/code-server/lib/node /var/tmp/coder/code-server
 \_ /var/tmp/coder/code-server/lib/node /var/tmp/coder/code-server
```
